File: src/main/java/com/uiptv/service/CacheService.java
package com.uiptv.service;

import com.uiptv.api.LoggerCallback;
import com.uiptv.model.Account;

import java.io.IOException;

public interface CacheService {

    default void clearAllCache() {
        ConfigurationService.getInstance().clearAllCache();
    }
    default void clearCache(Account account) {
        ConfigurationService.getInstance().clearCache(account);
    }

    void reloadCache(Account account, LoggerCallback logger) throws IOException;

    boolean verifyMacAddress(Account account, String macAddress, LoggerCallback logger);

    int getChannelCountForAccount(String accountId);
}

File: server/src/main/kotlin/com/uiptv/service/CacheService.kt
package com.uiptv.service

import com.uiptv.api.LoggerCallback
import com.uiptv.model.Account
import java.io.IOException

interface CacheService {
    fun clearAllCache() {
        ConfigurationService.clearAllCache()
    }

    fun clearCache(account: Account) {
        ConfigurationService.clearCache(account)
    }

    @Throws(IOException::class)
    fun reloadCache(account: Account, logger: LoggerCallback)

    fun verifyMacAddress(account: Account?, macAddress: String?): Boolean

    fun getChannelCountForAccount(accountId: String): Int
}
